src/dbsync/src/dbsyncPipelineFactory.cpp

```diff
@@ -0,0 +1,196 @@
+            (
+                std::bind(&Pipeline::processSyncRow, this, std::placeholders::_1),
+                threadNumber ? threadNumber : std::thread::hardware_concurrency()
+            );
+        }
+
+        SyncResult processSyncRow(const nlohmann::json& value)
+        {
+            ReturnTypeCallback type{ DB_ERROR };
+            nlohmann::json result{ value };
+            try
+            {
+                // DBSyncImplementation::instance().syncTxRow(m_handle, m_txnContext, value, type, result);
+            }
+            catch(const DbSync::max_rows_error& /*ex*/)
+            {
+                result = MAX_ROWS;
+            }
+            catch(...)
+            {
+                //we'll notify with DB_ERROR.
+            }
+            return std::make_tuple<ReturnTypeCallback, std::string>(std::move(type), result[0]);
+        }
+        void dispatchResult(const SyncResult& result)
+        {
+            const auto& value{ std::get<1>(result) };
+            if (!value.empty())
+            {
+                m_callback(std::get<0>(result), value);
+            }
+        }
+        const std::shared_ptr<DispatchCallbackNode> m_spDispatchNode;
+        const std::shared_ptr<SyncRowNode> m_spSyncNode;
+        const DBSYNC_HANDLE m_handle;
+        const TXN_HANDLE m_txnContext;
+        const unsigned int m_maxQueueSize;
+        const ResultCallback m_callback;
+    };
+//----------------------------------------------------------------------------------------
+    PipelineFactory& PipelineFactory::instance()
+    {
+        static PipelineFactory s_instance;
+        return s_instance;
+    }
+    void PipelineFactory::release()
+    {
+        std::lock_guard<std::mutex> lock{ m_contextsMutex };
+        m_contexts.clear();
+    }
+    PipelineCtxHandle PipelineFactory::create(const DBSYNC_HANDLE handle,
+                                              const char* tables,
+                                              const unsigned int threadNumber,
+                                              const unsigned int maxQueueSize,
+                                              const ResultCallback callback)
+    {
+        std::shared_ptr<IPipeline> spContext
+        {
+            new Pipeline
+            {
+                handle, tables, threadNumber, maxQueueSize, callback
+            }
+        };
+        const auto ret { spContext.get() };
+        std::lock_guard<std::mutex> lock{ m_contextsMutex };
+        m_contexts.emplace(ret, spContext);
+        return ret;
+    }
+    const std::shared_ptr<IPipeline>& PipelineFactory::pipeline(const PipelineCtxHandle handle)
+    {
+        std::lock_guard<std::mutex> lock{ m_contextsMutex };
+        const auto it
+        {
+            m_contexts.find(handle)
+        };
+        if (it == m_contexts.end())
+        {
+            throw dbsync_error
+            {
+                2, "PipelineFactory, Invalid handle value."
```

Comment:
you should change this by:
throw dbsync_error { INVALID_HANDLE }; 

Or create a new pair with ID and description and call it here.